src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -201,14 +201,7 @@ private async Task ProcessAsyncActionsV2(AsyncAction[] actions)
                 // to protect against possible exceptions. Exception handling
                 // and exception throwing is managed in the OOProc SDKs and
                 // should not affect / interrupt this replay loop
-                try
-                {
-                    await this.InvokeAPIFromAction(action);
-                }
-                catch (Exception)
-                {
-                    // ignore exceptions, they should be handled by user-code instead.
-                }
+                await Task.WhenAny(this.InvokeAPIFromAction(action));
```

Comment:
Thank you for this PR, this looks good to me; but I want to make sure @cgillum approves of the change as well.
In the meantime, could you please expand the comments above to justify why we use WhenAny instead of a try-catch? 

Otherwise, I could see a future maintainer removing the WhenAny for a try-catch again P . Let's break the cycle!

Reply:
Yeah that makes sense, I added the comment. Another idea I had was to wrap `this.InvokeAPIFromAction()` in a separate custom class/Task that we create, rather than a `WhenAny` to make the intentions clearer

Reply:
I don't have any problem with this change but I'll admit I still don't understand how or why it works. For example, we say that WhenAny swallows exceptions, but since that's not the documented purpose of the `Task.WhenAny` API, it's not clear to me why this is the appropriate thing to do. It's also not clear to me why "swallowing exceptions" is a good thing; we mention that it's the SDKs responsibility to handle exceptions, but how is that related to the tasks in the C# layer?

Long story short, it feels like there's a lot of magic going on here and some context and it's not clear to me how easy it will be for future maintainers (or our futures selves) to understand what the intent of this code is if we find another problem with it later. More comments explaining more of the mechanics may help. However, I don't necessarily want to block on this issue - it's more of a suggestion.

Reply:
Let's make sure we remove the magic here, I think that's crucial for these OOProc bits. I'll coordinate with Hossam to add a few more comments, and I'll loop you back at that point. Thanks!

Reply:
@cgillum After talking with @davidmrdavid I think we found a solution that achieves the same affect while using less "magic." I've updated the PR, let me know how this solution works now :)